pym/bob/scm/url.py

```diff
@@ -98,24 +96,49 @@ def asScript(self):
 cp -uT {URL} {FILE}
 """
         else:
+            # do command logic inside the shell script to leave it to the backend
             tpl = """
 {HEADER}
 mkdir -p {DIR}
 cd {DIR}
 if [ -e {FILE} ] ; then
-    curl {OPTIONS} -o {FILE} -z {FILE} {URL}
+    if [ -x "$(command -v curl)" ] ; then
+        OPTIONS="-sSgLf"
```

Comment:
If you set variables you will have to put that part in a sub-shell to prevent leaking the variable in the `checkoutScript` that is possibly concatenated. But as the list of options is fixed I would rather pass it directly into the script, e.g. as `{CURL_OPTIONS}` and `{WGET_OPTIONS}`.